compiler/rustc_interface/src/passes.rs

```diff
@@ -746,81 +746,69 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
     rustc_passes::hir_id_validator::check_crate(tcx);
 
     let sess = tcx.sess;
-    let mut entry_point = None;
 
-    sess.time("misc_checking_1", || {
-        parallel!(
-            {
-                entry_point = sess.time("looking_for_entry_point", || tcx.entry_fn(()));
+    parallel!(
+        {
+            tcx.ensure().entry_fn(());
 
-                sess.time("looking_for_derive_registrar", || {
-                    tcx.ensure().proc_macro_decls_static(())
-                });
+            tcx.ensure().proc_macro_decls_static(());
 
-                CStore::from_tcx(tcx).report_unused_deps(tcx);
-            },
-            {
-                tcx.hir().par_for_each_module(|module| {
-                    tcx.ensure().check_mod_loops(module);
-                    tcx.ensure().check_mod_attrs(module);
-                    tcx.ensure().check_mod_naked_functions(module);
-                    tcx.ensure().check_mod_unstable_api_usage(module);
-                    tcx.ensure().check_mod_const_bodies(module);
-                });
-            },
-            {
-                sess.time("unused_lib_feature_checking", || {
-                    rustc_passes::stability::check_unused_or_stable_features(tcx)
-                });
-            },
-            {
-                // We force these queries to run,
-                // since they might not otherwise get called.
-                // This marks the corresponding crate-level attributes
-                // as used, and ensures that their values are valid.
-                tcx.ensure().limits(());
-                tcx.ensure().stability_index(());
-            }
-        );
-    });
+            CStore::from_tcx(tcx).report_unused_deps(tcx);
+        },
+        {
+            tcx.hir().par_for_each_module(|module| {
+                tcx.ensure().check_mod_loops(module);
+                tcx.ensure().check_mod_attrs(module);
+                tcx.ensure().check_mod_naked_functions(module);
+                tcx.ensure().check_mod_unstable_api_usage(module);
+                tcx.ensure().check_mod_const_bodies(module);
+            });
+        },
+        {
+            rustc_passes::stability::check_unused_or_stable_features(tcx);
+        },
+        {
+            // We force these queries to run,
+            // since they might not otherwise get called.
+            // This marks the corresponding crate-level attributes
+            // as used, and ensures that their values are valid.
+            tcx.ensure().limits(());
+            tcx.ensure().stability_index(());
+        }
+    );
 
     // passes are timed inside typeck
     rustc_hir_analysis::check_crate(tcx)?;
 
-    sess.time("MIR_borrow_checking", || {
-        tcx.hir().par_body_owners(|def_id| tcx.ensure().mir_borrowck(def_id));
-    });
```

Comment:
This removes the aggregate count of borrow checking time from `-Ztime-passes`, right? AFAIK queries are not shown by `-Ztime-passes`.

Reply:
Yes. That would remove it.
I wonder if we should remove `-Ztime-passes` entirely. It's already incorrect (some borrow-check may already have happened earlier), and bound to be wronger with parallel compilation.

Reply:
`-Ztime-passes` is much more convenient when trying to find roughly where the bottleneck is. I don't see myself asking someone to install measureme to get a rough idea why rustc is so slow on their code. Also `-Zself-profile` only works if rustc terminates in the first place. If it gets stuck somewhere `-Zself-profile` is the only thing that can provide a guess where it is stuck.

Reply:
what we could do here is

* use `tracing-indicatif` for showing progress on queries as the compiler runs
* use a custom tracing subscriber that accumulates data for every query and how long it was run and emits that information at the end.

But I think that's orthogonal to this PR. We've already reduced the usefulness of `-Ztime-passes` for "phase tracking" in previous PRs that did these kind of changes and as @cjgillot said, we've already made it less useful years ago just by introducing queries.